File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> mp;
        int max = 0, num;
        int majority = (nums.size() / 2);
        for(auto n : nums){
            mp[n]++;
        }
        map<int, int>::iterator it;
        for (it = mp.begin(); it != mp.end(); it++)
        {
            if(it -> second >= majority)
            {
                if (it -> second > max){
                    max = it -> second;
                    num = it -> first;
                }

            }
        }
        return num;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> mp = new HashMap<>();
        int max = 0, num = 0;
        int majority = (nums.length / 2);
        for(int n : nums){
            if (mp.containsKey(n)) {
                mp.put(n, mp.get(n) + 1); 
            } else {
                mp.put(n, 1); 
            }
        }
        for (Map.Entry<Integer, Integer> it : mp.entrySet())
        {
            if(it.getValue() >= majority)
            {
                if (it.getValue() > max){
                    max = it.getValue();
                    num = it.getKey();
                }

            }
        }
        return num;
    }
}